libnavigation-ui/src/test/java/com/mapbox/navigation/ui/route/MapRouteLineTest.kt

```diff
@@ -1124,18 +1112,18 @@ class MapRouteLineTest {
             "[\"rgba\", 0.0, 0.0, 0.0, 0.0], 0.01792979, [\"rgba\", 47.0, 122.0, 198.0, 1.0]]"
         val expectedRouteTrafficLineVanishingExpression = "[\"step\", [\"line-progress\"], " +
             "[\"rgba\", 0.0, 0.0, 0.0, 0.0], 0.01792979, " +
-            "[\"rgba\", 86.0, 168.0, 251.0, 1.0], 0.056129422, " +
-            "[\"rgba\", 86.0, 168.0, 251.0, 1.0], 0.09373502, " +
-            "[\"rgba\", 86.0, 168.0, 251.0, 1.0], 0.150941, " +
-            "[\"rgba\", 86.0, 168.0, 251.0, 1.0], 0.1905395, " +
-            "[\"rgba\", 86.0, 168.0, 251.0, 1.0], 0.26964808, " +
-            "[\"rgba\", 255.0, 77.0, 77.0, 1.0], 0.27944908, " +
-            "[\"rgba\", 86.0, 168.0, 251.0, 1.0], 0.33272266, " +
-            "[\"rgba\", 86.0, 168.0, 251.0, 1.0], 0.39298487, " +
-            "[\"rgba\", 86.0, 168.0, 251.0, 1.0], 0.48991048, " +
-            "[\"rgba\", 255.0, 77.0, 77.0, 1.0], 0.504132, " +
-            "[\"rgba\", 86.0, 168.0, 251.0, 1.0], 0.8017851, " +
-            "[\"rgba\", 86.0, 168.0, 251.0, 1.0], 1.0000086, [\"rgba\", 86.0, 168.0, 251.0, 1.0]]"
+            "[\"rgba\", 86.0, 168.0, 251.0, 1.0], 0.05612764, " +
+            "[\"rgba\", 86.0, 168.0, 251.0, 1.0], 0.10589197, " +
+            "[\"rgba\", 86.0, 168.0, 251.0, 1.0], 0.15093622, " +
+            "[\"rgba\", 86.0, 168.0, 251.0, 1.0], 0.20669733, " +
```

Comment:
We should able to filter out these color duplicates, right? At this point, we are not concerned about the congestion level anymore, but just about the color we actually assign. This might significantly simplify the expression for longer routes and improve performance even more.

Let's do that as a follow-up.

Reply:
done